src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Tables/ResourceChangeData.sql

```diff
@@ -6,6 +6,8 @@ CREATE TABLE dbo.ResourceChangeData
     ResourceId varchar(64) NOT NULL,
     ResourceTypeId smallint NOT NULL,
     ResourceVersion int NOT NULL,
-    ResourceChangeTypeId tinyint NOT NULL,
-    CONSTRAINT PK_ResourceChangeData_TimestampId PRIMARY KEY (Timestamp, Id)
-) ON PartitionScheme_ResourceChangeData_Timestamp(Timestamp)
+    ResourceChangeTypeId tinyint NOT NULL
+) ON PartitionScheme_ResourceChangeData_Timestamp(Timestamp);
+
+CREATE CLUSTERED INDEX IXC_ResourceChangeData ON dbo.ResourceChangeData
+    (Id ASC) WITH(ONLINE = ON) ON PartitionScheme_ResourceChangeData_Timestamp(Timestamp);
```

Comment:
Is there a reason, we are not creating primary key on Id column?

Reply:
Yes, the reason is to have a better performance on the fetch query. Also, since a resourceChangeData table is partitioned on timestamp, we can not create the primary key only on the Id column due to a SQL constraint, "partition columns for a unique index must be a subset of the index key".

Previously, the uniqueness was combined with the timestamp column and it was only per partition. To enforce global uniqueness requires a non clustered index without a partition but which prevents partition swaps.

We are using identity which will guarantee uniqueness unless an identity insert is used or reseed identity value on the table which shouldn't happen.

Reply:
Added a comment.